services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,880 @@
+package org.openkilda.functionaltests.spec.flows
+
+import static org.junit.Assume.assumeTrue
+import static org.openkilda.testing.Constants.NON_EXISTENT_FLOW_ID
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.error.MessageError
+import org.openkilda.messaging.info.event.IslChangeType
+import org.openkilda.messaging.info.event.PathNode
+import org.openkilda.messaging.payload.flow.FlowState
+import org.openkilda.model.Cookie
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Narrative
+import spock.lang.Unroll
+
+@Narrative("""Protected path - it is pre-calculated, reserved, and deployed (except ingress rule),
+so we can switch traffic fast.
+
+- flow object is extended with a boolean parameter 'allocate_protected_path' with values false(default)
+- /flows/{flow-id}/path returns 'main_path' + 'protected_path'.
+
+System can start to use protected path in two case:
+- main path is down;
+- we send the 'swap' request for a flow with protected path('/v1/flows/{flow_id}/swap')
+
+Main and protected paths can't use the same link.""")
+class ProtectedPathSpec extends BaseSpecification {
+
+    @Value('${reroute.hardtimeout}')
+    int rerouteHardtimeout
+
+    @Unroll
+    def "Able to create flow with 'protected path' in case maximumBandwidth=#bandwidth, vlan=#vlanId"() {
```

Comment:
in case of ...

Reply:
I will try to avoid this combination in my tests
will rename `in case -> when`